demos/python_demos/image_retrieval_demo/image_retrieval_demo/visualizer.py

```diff
@@ -95,6 +95,8 @@ def visualize(image, target_pos, impaths, distances, input_size, compute_embeddi
 
     demo_image = add_top10_gallery_images(demo_image, impaths, distances, input_image, target_pos)
 
-    cv2.imshow('demo_image', demo_image)
+    if not no_show:
+        cv2.imshow('demo_image', demo_image)
+        return cv2.waitKey(imshow_delay) & 0xff
 
-    return cv2.waitKey(imshow_delay) & 0xff
+    return 0
```

Comment:
It should be `-1`, like if no key had been pressed: https://www.docs.opencv.org/master/d7/dfc/group__highgui.html#ga5628525ad33f52eab17feebcfba38bd7

Reply:
And `-1 & 0xff` is probably something different.

Reply:
@eizamaliev, so what about this `0xff`? `-1 & 0xff` which you will get from `cv2.waitKey(imshow_delay) & 0xff` is not equal to `-1` in case of `--no_show`. We could just drop this `0xff` from the demo or return `-1 & 0xff`. I don't know which way is correct to process pressed key.

Reply:
In this case `-1` will be correct output for no-key-pressed. In fact, `cv2.waitKey(imshow_delay) & 0xff` is incorrect output if no key pressed. I'll fix it.

Reply:
`cv::waitKey` returns not only -1 or 1.
Your line `return key_pressed & 0xff if key_pressed != 1 else -1` is equivalent to 
```
if key_pressed == 1: return -1
else return key_pressed & 0xff
```
So if you press the button with code `1`, the demo will receive `-1` and since pressed key is handled out of the function, I believe it's better to forward code of pressed key unchanged (the demo doesn't handle that particular key now, but I still think it is incorrect approach). Moreover for current implementation, if you press nothing and presenting is enabled, you will get `-1 & 0xff` and if you use `--no_show` it will return `-1`, while my suggestion is that `--no_show` should behave like no key is pressed.

Reply:
> In this case `-1` will be correct output for no-key-pressed. In fact, `cv2.waitKey(imshow_delay) & 0xff` is incorrect output if no key pressed. I'll fix it.

In your implementation for no-key-pressed the return value will be `-1 & 0xff` which is not `-1`.

Reply:
Sorry, my mistake. It should have been *key_pressed != **-1***

Reply:
I not sure, that understand you, but will summarize:

Key | with `--no_show` | without `--no_show 
----- | ----- | -----
| Pressed | -1 | waitKey() & 0xFF |
| Not pressed| -1 | -1 (not waitKey() & 0xFF) |

Correct?

Reply:
@Wovchena do you accept this change?

Reply:
Yes